lib/ansible/modules/packaging/language/pip.py

```diff
@@ -682,6 +702,8 @@ def main():
             cmd.extend(to_native(p) for p in packages)
         elif requirements:
             cmd.extend(['-r', requirements])
+            if constraints:
```

Comment:
constraints isn't dependent on requirements.  constraints can still be useful if the libraries being installed are liberal in their selection.

This should be moved outside.